src/__spec__/when.spec.ts

```diff
@@ -0,0 +1,16 @@
+import assert from 'static-type-assert';
+import { when } from '..';
+
+declare var Ø: never;
```

Comment:
Fancy, but I would prefer English.

Reply:
I couldn’t think of a good name for something that doesn’t exist. I wanted a really short name because this is used across the type specs now. I thought of a names like `any` or `never`, but those are the names of types, and would easily be confused with the actual types when trying to read the code. I liked Ø because it’s short, can mean “the empty set” in mathematical notation, and is visually distinct, which draws attention to positional parameters, sort of like a comma.

Reply:
Why didn't you do this?

```typescript
declare const ANY: any;

assert<Iterable<number>>(when(ANY, [] as Iterable<number>));

assert<[1, 2, 3] | []>(when(ANY, [1, 2, 3] as [1, 2, 3]));

assert<Object>(when(ANY, ANY as Object));

assert<{test: 'test'} | {}>(when(ANY, {test: 'test' as 'test'}));

assert<{}>({...when(ANY, null)});

assert<Array<never>>([...when(ANY, null)]);
```

Reply:
My pattern seemed simpler? There’s only one thing that occurs in a value position, and all type annotations are completely explicit, which is exactly what I want in a type test.

Reply:
Also I gravitate towards rules that I can explain as simply as possible, given that I hope that eventually this project will no longer require a significant amount of my time to maintain. I could easily write a lint rule that says in spec files only `Ø` should go in the value position. That would guarantee that it is the responsibility of the `__spec__/*.spec.ts` files to test type evaluation only. That way no-one is tempted to do things like use `1` as a shorthand for `number`, not be confused with `1 as 1`.